cleverpush/src/main/java/com/cleverpush/CleverPush.java

```diff
@@ -1692,146 +1754,146 @@ public void onFailure(int statusCode, String response, Throwable throwable) {
-								editor.remove(CleverPushPreferences.SUBSCRIPTION_ATTRIBUTES).apply();
-								editor.putString(CleverPushPreferences.SUBSCRIPTION_ATTRIBUTES, jsonString);
-								editor.commit();
-							}
-						} catch (Exception ex) {
-							Log.e("CleverPush", ex.getMessage(), ex);
-						}
-					}
-
-					@Override
-					public void onFailure(int statusCode, String response, Throwable throwable) {
-						Log.e("CleverPush", "Error pulling attribute value - HTTP " + statusCode);
-					}
-				});
-			}
-		})).start());
-	}
-
-	public boolean hasSubscriptionAttributeValue(String attributeId, String value) {
-		Map<String, Object> subscriptionAttributes = this.getSubscriptionAttributes();
-
-		ArrayList<String> arrayList = new ArrayList<>();
-		try {
-			JSONArray arrayValue = (JSONArray) subscriptionAttributes.get(attributeId);
-			if (arrayValue != null) {
-				for (int i = 0; i < arrayValue.length(); i++) {
-					arrayList.add(arrayValue.getString(i));
-				}
-			}
-		} catch (Exception ex) {
-			Log.e("CleverPush", ex.getMessage(), ex);
-		}
-
-		return arrayList.contains(value);
-	}
+    public void pushSubscriptionAttributeValue(String attributeId, String value) {
+        this.waitForTrackingConsent(() -> new Thread(() -> this.getSubscriptionId(subscriptionId -> {
+            if (subscriptionId != null) {
+                JSONObject jsonBody = new JSONObject();
+                try {
+                    jsonBody.put("channelId", this.channelId);
+                    jsonBody.put("attributeId", attributeId);
+                    jsonBody.put("value", value);
+                    jsonBody.put("subscriptionId", subscriptionId);
+                } catch (JSONException ex) {
+                    Log.e("CleverPush", ex.getMessage(), ex);
+                }
+
+                SharedPreferences sharedPreferences = PreferenceManager.getDefaultSharedPreferences(CleverPush.context);
+                Map<String, Object> subscriptionAttributes = this.getSubscriptionAttributes();
+
+                ArrayList<String> arrayList = new ArrayList<>();
+                try {
+                    JSONArray arrayValue = (JSONArray) subscriptionAttributes.get(attributeId);
+                    if (arrayValue != null) {
+                        for (int i = 0; i < arrayValue.length(); i++) {
+                            arrayList.add(arrayValue.getString(i));
+                        }
+                    }
+                } catch (Exception ex) {
+                    Log.e("CleverPush", ex.getMessage(), ex);
+                }
+
+                arrayList.add(value);
+                String[] arrayString = arrayList.toArray(new String[0]);
+                if (arrayString == null) {
+                    return;
+                }
+                subscriptionAttributes.put(attributeId, arrayString);
+
+                CleverPushHttpClient.post("/subscription/attribute/push-value", jsonBody, new CleverPushHttpClient.ResponseHandler() {
```

Comment:
please export new CleverPushHttpClient.ResponseHandler() to a method, returning a new anonymous instance of ResponseHandler